frontend/src/utils/server-utils.ts

```diff
@@ -395,20 +394,12 @@ export function generateIntermediateDateItemFromValidity(
         endDate: endDate.getTime(),
       }));
 
-      // We filter the dates that don't include the displayDate of the previous item array
-      const filteredDateItems = acc.filter(
-        dateItem => !binaryIncludes(daysToAdd, dateItem.displayDate, x => x),
-      );
-
-      return [...filteredDateItems, ...dateItemsToAdd];
+      return [...acc, ...dateItemsToAdd];
     }, []);
 
   // We sort the defaultDateItems and the dateItemsWithValidity and we order by displayDate to filter the duplicates
   // or the overlapping dates
-  return sortedUniqBy(
-    sortBy(dateItemsWithValidity, 'displayDate'),
-    'displayDate',
-  );
+  return sortBy(dateItemsWithValidity, 'displayDate');
```

Comment:
Removing these deduping steps does fix #1266. This update essentially increases the number of "available dates" to handle the overlapping periods. Before this change, when a date was found to be within two ranges, it'd be filtered from our list. With this change, that same date would in our list of available dates under all valid ranges.

This does _function_ as we'd desire but it does so by inflating our date list by a factor equaling the dekad in the layer's validity range. For "months", our list of dates to run through increases by 3x. For our seasons, it increases by 9x.

Creating this list doesn't result in any real slowdown of our date initialization. **However**, renders in react are slowed by operations which run through this list (particularly running the `datesAreEqualWithoutTime` function). I'm seeing a slowdown of about half a second in the first few renders on the page.

| Before | After |
| - | - |
| <img width="788" alt="Screenshot 2024-07-13 at 1 06 46 PM" src="https://github.com/user-attachments/assets/d0899266-aa7e-440f-89e6-170a2dcb5fa6"> | <img width="930" alt="Screenshot 2024-07-13 at 12 59 54 PM" src="https://github.com/user-attachments/assets/31bc86d5-ce7f-4821-9d2a-7831b78b4f6a"> |

I'm pursuing two solutions:
- Can we speed up the `datesAreEqualWithoutTime` function by using date-fns so the larger lists aren't an issue?
- Can we base validity time window off of the start and end times of the selected time rather than reading from from the list of available dates? That removes the need to inflate this list

Reply:
I was able to speed up datesAreEqualWithoutTime by 1000x by updating to the function to use math rather than date objects (at least in Chrome browsers).
<img width="919" alt="Screenshot 2024-07-14 at 3 18 26 PM" src="https://github.com/user-attachments/assets/94c5ed03-535e-4d65-9193-9f4bcec596ad">

This brings the performance back up to where things were before. However, I do still wonder if it makes sense to store validity dates in "available dates". I think I'd prefer to keep available dates as our valid dekad options (1st, 11th, 21st) and rely on the start and end dates provide to determine validity at render rather than fill our available dates with multiples of validity dates. I'm going to explore what this refactor would look like

Reply:
Before refactoring validity dates our "available dates", we're already at performance improvements on what existed before. I'm going to hold off on that refactor for now
<img width="481" alt="Screenshot 2024-07-15 at 10 16 57 AM" src="https://github.com/user-attachments/assets/5104f5f8-f8a0-4ca6-b09e-0698f9941143">